homeassistant/components/glances/coordinator.py

```diff
@@ -42,4 +44,19 @@ async def _async_update_data(self) -> dict[str, Any]:
             raise ConfigEntryAuthFailed from err
         except exceptions.GlancesApiError as err:
             raise UpdateFailed from err
+        # Update computed values
+        if data:
+            up_duration = parse_duration(data.get("uptime"))
+            uptime = None
+            if isinstance(up_duration, timedelta):
+                # Update uptime if previous value is None or previous uptime is bigger than
+                # new uptime (i.e. server restarted) else use previous value
+                if (
+                    self.data is None
+                    or self.data["computed"]["uptime_duration"] > up_duration
+                ):
+                    uptime = utcnow() - up_duration
+                elif self.data:
+                    uptime = self.data["computed"]["uptime"]
+            data["computed"] = {"uptime_duration": up_duration, "uptime": uptime}
```

Comment:
```suggestion
        # Update computed values
        uptime: datetime | None = self.data["computed"]["uptime"] if self.data else None
        up_duration: timedelta | None = None
        if data and (up_duration := parse_duration(data.get("uptime"))):
            # Update uptime if previous value is None or previous uptime is bigger than
            # new uptime (i.e. server restarted) else use previous value
            if (
                self.data is None
                or self.data["computed"]["uptime_duration"] > up_duration
            ):
                uptime = utcnow() - up_duration
        data["computed"] = {"uptime_duration": up_duration, "uptime": uptime}
```
Something like this perhaps to reduce the code a bit?

Reply:
Yes looks good, done.